dd-java-agent/agent-iast/src/main/java/com/datadog/iast/sink/InsecureCookieModuleImpl.java

```diff
@@ -1,28 +1,41 @@
 package com.datadog.iast.sink;
 
 import com.datadog.iast.model.Evidence;
+import com.datadog.iast.model.Location;
+import com.datadog.iast.model.Vulnerability;
 import com.datadog.iast.model.VulnerabilityType;
 import com.datadog.iast.overhead.Operations;
+import com.datadog.iast.util.CookieSecurityInfo;
 import datadog.trace.api.iast.sink.InsecureCookieModule;
 import datadog.trace.bootstrap.instrumentation.api.AgentSpan;
 import datadog.trace.bootstrap.instrumentation.api.AgentTracer;
-import javax.annotation.Nonnull;
 
-public class InsecureCookieModuleImpl extends SinkModuleBase implements InsecureCookieModule {
+public class InsecureCookieModuleImpl extends SinkModuleBase
+    implements InsecureCookieModule, ForCookieSecurityInfo {
 
   @Override
-  public void onCookie(
-      @Nonnull final String name,
-      final String value,
-      final boolean isSecure,
-      final boolean isHttpOnly,
-      final String sameSite) {
-    if (!isSecure) {
-      final AgentSpan span = AgentTracer.activeSpan();
-      if (!overheadController.consumeQuota(Operations.REPORT_VULNERABILITY, span)) {
-        return;
+  public void onCookie(CookieSecurityInfo cookieSecurityInfo) {
```

Comment:
I think we should reuse most of the onCookie code as `InsecureCookieModuleImpl`, `NoHttpOnlyCookieModuleImpl` and `NoSameSiteCookieModuleImpl` implementations are pretty close.

```suggestion
  public void onCookie(CookieSecurityInfo cookieSecurityInfo) {
  if (!cookieSecurityInfo.isSecure()) {
  checkCookieVulnerability(VulnerabilityType.INSECURE_COOKIE)
  }
  }
```